### DISCUSSION TITLE
Connect to FTP(S) -Server

### DISCUSSION THREAD

**Discussion by Tueftel99** [wqweto/VbAsyncSocket#31] [General]
Hello Vladimir,
How do I start to access an FTP(S) or SFTP server with VB6.

A short introduction (example) would be nice.

greeting

**Comment by wqweto** [comment]
SFTP has nothing to do with FTPS. SFTP is an extension to SSH and cannot be implemented w/ this repo while FTPS is based on TLS so it's possible to use this repo's classes to implement it.

FTPS is to FTP what HTTPS is to HTTP i.e. FTP and FTPS both use the same protocol but w/ FTPS the client initiates/negotiates TLS protected connection to the server.

FTPS comes in two variants: explicit and implicit. The implicit variant is very much like HTTPS, it has a dedicated 990 port and the connection is secured with TLS from the beginning. Unfortunately implicit FTPS is deprecated so the only recommended variant is explicit FTPS also called FTPES.

With FTPES the client starts a normal unprotected connection on port 21 and immediately negotiates TLS with `AUTH TLS` command which initiates TLS negotiation on the same connection.

In repo's README there is already a sample code for `STARTTLS` command with SMTP protocol to gmail.com server so the idea of FTPES is basically the same.
